internal/builders/docker/commands.go

```diff
@@ -69,23 +68,34 @@ func writeBuildDefinitionToFile(bd pkg.BuildDefinition, path string) error {
 // BuildCmd builds the artifacts using the input flags, and prints out their digests, or exists with an error.
 func BuildCmd(check func(error)) *cobra.Command {
 	io := &pkg.InputOptions{}
+	var forceCheckout bool
 
 	cmd := &cobra.Command{
 		Use:   "build [FLAGS]",
 		Short: "Builds the artifacts using the build config, source repo, and the builder image.",
 		Run: func(cmd *cobra.Command, args []string) {
 			config, err := pkg.NewDockerBuildConfig(io)
 			check(err)
-			log.Printf("The config is: %v\n", config)
 
-			// TODO(#1191): Set up build state using config, and build the artifact.
-			artifacts := "To be implemented"
+			builder, err := pkg.NewBuilderWithGitFetcher(*config, forceCheckout)
+			check(err)
+			db, err := builder.SetUpBuildState()
+			check(err)
+			artifacts, err := db.BuildArtifact()
+			check(err)
 			log.Printf("Generated artifacts are: %v\n", artifacts)
-			// TODO(#1191): Write subjects to file.
+			if db.RepoInfo.RepoRoot != "" {
+				// If the repo was checked out by the tool into a temporary
+				// directory, remove the files and the directory.
+				db.RepoInfo.Cleanup()
```

Comment:
can you run this as a `defer db.RepoInfo.Cleanup()` just in case an error occurs that causes this function to exit early? I think you can always run it, since the `db.RepoInfo.RepoRoot` check is internal to the cleanup too.

Reply:
Very good point. Done.